examples/misc/demo_agg_filter.py

```diff
@@ -4,63 +4,52 @@
 ===============
 
 """
-import matplotlib.pyplot as plt
 
-import numpy as np
 import matplotlib.cm as cm
+import matplotlib.pyplot as plt
 import matplotlib.transforms as mtransforms
 from matplotlib.colors import LightSource
 from matplotlib.artist import Artist
+import numpy as np
 
 
 def smooth1d(x, window_len):
     # copied from http://www.scipy.org/Cookbook/SignalSmooth
-
     s = np.r_[2*x[0] - x[window_len:1:-1], x, 2*x[-1] - x[-1:-window_len:-1]]
     w = np.hanning(window_len)
     y = np.convolve(w/w.sum(), s, mode='same')
     return y[window_len-1:-window_len+1]
 
 
 def smooth2d(A, sigma=3):
-
-    window_len = max(int(sigma), 3)*2 + 1
-    A1 = np.array([smooth1d(x, window_len) for x in np.asarray(A)])
-    A2 = np.transpose(A1)
-    A3 = np.array([smooth1d(x, window_len) for x in A2])
-    A4 = np.transpose(A3)
-
-    return A4
+    window_len = max(int(sigma), 3) * 2 + 1
+    A = np.apply_along_axis(smooth1d, 0, A, window_len)
+    A = np.apply_along_axis(smooth1d, 1, A, window_len)
+    return A
 
 
 class BaseFilter:
-    def prepare_image(self, src_image, dpi, pad):
-        ny, nx, depth = src_image.shape
-        # tgt_image = np.zeros([pad*2+ny, pad*2+nx, depth], dtype="d")
-        padded_src = np.zeros([pad*2 + ny, pad*2 + nx, depth], dtype="d")
-        padded_src[pad:-pad, pad:-pad, :] = src_image[:, :, :]
-
-        return padded_src  # , tgt_image
 
     def get_pad(self, dpi):
         return 0
 
+    def process_image(padded_src, dpi):
+        raise NotImplementedError("Should be overridden by subclasses")
+
     def __call__(self, im, dpi):
         pad = self.get_pad(dpi)
-        padded_src = self.prepare_image(im, dpi, pad)
+        padded_src = np.pad(im, [(pad, pad), (pad, pad), (0, 0)], "constant")
         tgt_image = self.process_image(padded_src, dpi)
         return tgt_image, -pad, -pad
 
 
 class OffsetFilter(BaseFilter):
-    def __init__(self, offsets=None):
-        if offsets is None:
-            self.offsets = (0, 0)
-        else:
-            self.offsets = offsets
+
+    def __init__(self, offsets=(0, 0)):
+        self.offsets = offsets
 
     def get_pad(self, dpi):
-        return int(max(*self.offsets)/72.*dpi)
+        return int(max(self.offsets)/72.*dpi)
```

Comment:
Since you're changing this line, you can remove the dot and add spacing.

Reply:
sure